File: TsogosunProfileAdmin/MSProfileAdmin/Service/CountryService.cs

using System.Collections.Generic;
using tsogosun.com.MSProfileAdmin.Model.Dtos;
using tsogosun.com.MSProfileAdmin.Repository.Interface;
using tsogosun.com.MSProfileAdmin.Service.Interface;

namespace tsogosun.com.MSProfileAdmin.Service
{
    public class CountryService : ICountryService
    {

        private readonly ICountryRepository _countryRepository;

        public CountryService(ICountryRepository countryRepository)
        {
            _countryRepository = countryRepository;
        }

        public List<CountryDto> GetCountriesByName(string countryName)
        {
            return _countryRepository.GetCountriesByName(countryName);
        }

        public List<CountryDto> GetCountries()
        {
            return _countryRepository.GetCountries();
        }       

        public List<ProvinceDto> GetProvincesByName(string provinceName)
        {
            return _countryRepository.GetProvincesByName(provinceName);
        }

        public List<ProvinceDto> GetProvinces()
        {
            return _countryRepository.GetProvinces();
        }

        public List<PostalCodeDto> GetPostalCodesByCode(string postalCode)
        {
            return _countryRepository.GetPostalCodesByCode(postalCode);
        }

        public List<CityDto> GetCityByName(string cityName)
        {
            return _countryRepository.GetCityByName(cityName);
        }

        public List<SuburbDto> GetSuburbByName(string suburbName)
        {
            return _countryRepository.GetSuburbByName(suburbName);
        }
    }
}

File: TsogosunProfileAdminWeb/src/app/service/profile-admin/country.service.ts
import { Injectable } from '@angular/core';
import { CityDto } from 'src/app/model/Dtos/profile-admin/cityDto';
import { CountryDto } from 'src/app/model/Dtos/profile-admin/countryDto';
import { PostalCodeDto } from 'src/app/model/Dtos/profile-admin/postalCodeDto';
import { ProvinceDto } from 'src/app/model/Dtos/profile-admin/provinceDto';
import { SuburbDto } from 'src/app/model/Dtos/profile-admin/suburbDto';
import { BackendService } from 'src/app/util/backend.service';

@Injectable({
    providedIn: 'root'
})
export class CountryService {

    constructor(private backend: BackendService) {

    }

    async getCountries() {
        return await this.backend.get<CountryDto[]>(`Country/GetAll`);
    }

    async getProvincesByName(name: string) {
        return await this.backend.get<ProvinceDto[]>(`Country/Province/Get/${name}`);
    }

    async getPostalCodesByCode(name: string) {
        return await this.backend.get<PostalCodeDto[]>(`Country/PostalCode/Get/${name}`);
    }

    async getCityByName(name: string) {
        return await this.backend.get<CityDto[]>(`Country/City/Get/${name}`);
    }

    async getSuburbByName(name: string) {
        return await this.backend.get<SuburbDto[]>(`Country/Suburb/Get/${name}`);
    }
}
